presto-main/src/main/java/com/facebook/presto/sql/planner/CanonicalPlanGenerator.java

```diff
@@ -312,18 +366,40 @@ public Optional<PlanNode> visitTableScan(TableScanNode node, Map<VariableReferen
             outputVariables.add(reference);
             assignments.put(reference, columnReference.getColumnHandle());
         }
+        TupleDomain<ColumnHandle> constraint = node.getCurrentConstraint();
+
+        switch (strategy) {
+            case DEFAULT:
+                // This setting is used for fragment result caching, and does not need current constraint.
+                constraint = TupleDomain.all();
+                break;
+            case CONNECTOR:
```

Comment:
maybe I need to understand the meaning of `CONNECTOR` strategy. What does that mean?

Reply:
answered above